src/network/serverpackethandler.cpp

```diff
@@ -1821,3 +1820,30 @@ void Server::handleCommand_ModChannelMsg(NetworkPacket *pkt)
 
 	broadcastModChannelMessage(channel_name, channel_msg, peer_id);
 }
+
+void Server::handleCommand_HaveMedia(NetworkPacket *pkt)
+{
+	std::vector<u32> tokens;
+	u8 numtokens;
+
+	*pkt >> numtokens;
+	for (u16 i = 0; i < numtokens; i++) {
+		u32 n;
+		*pkt >> n;
+		tokens.emplace_back(n);
+	}
+
+	const session_t peer_id = pkt->getPeerId();
+	auto player = m_env->getPlayer(peer_id);
```

Comment:
Hmm, are you sure?

I only retrieve the name of the sender once, other clients are not touched and if other functions run from inside Lua they will do their own locking.

Reply:
You're retrieving the name in every iteration where count > 0. The callbacks take a while to execute; giving async tasks plenty of time to modify the client list. That was the idea. However since any other player action (i.e. leaving) is processed by callbacks as well, this seems to be fine on a second thought.